### PR TITLE
chore: v23 release notes

### PR DESCRIPTION
**Pull request by timhuynh94**

### CODE REVIEW

File: releases/v0.23.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+### 📣 Highlights
+
+
+### 💥 Breaking Changes
+
+#### Admins:
+- `v0.23.x` introduces a new `deployments` table that will be created automatically by default. However, if you are running the `server` component with the `VELA_DATABASE_SKIP_CREATION` set to `true`, you will need to manually create this table.
+
+  - https://github.com/go-vela/server/pull/1031
+
```
Comment (wass3rw3rk) [inline releases/v0.23.md:21]: We should add a Users section that talks about the change to external contributor PRs (forks). That newly created repos will default to require someone with write permissions to allow PRs to run.

File: releases/v0.23.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
+# v0.23 🚀
+
+This document contains all release notes pertaining to the `v0.23.x` releases of
+Vela.
+
+## For Vela Administrators
+
+- When upgrading from `v0.23`, please take note of the [migration information](/migrations/v0.23/deploy-recovery.md)
+
```
Comment (wass3rw3rk) [inline releases/v0.23.md:9]: We need to add the other migration information regarding SQL queries that need to be ran? Maybe after https://github.com/go-vela/community/pull/906 merged?

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]
few comments.. did you use the script to generate most of the release notes? i was sure it sorted alphabetically before?!

File: releases/v0.23.md (hunk 1 above)
Comment (timhuynh94) [reply releases/v0.23.md]: Added

File: releases/v0.23.md (hunk 2 above)
Comment (timhuynh94) [reply releases/v0.23.md]: I referenced it and waiting for #906  to be merged

File: releases/v0.23.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+- (server) feat(repo)!: support allow_events struct [#1023](https://github.com/go-vela/server/commit/6fcde8652f63e317137a63cb4f01586bbbbc49c2) - thanks [@ecrupper](https://github.com/ecrupper)!
+- (server) feat(webhook)!: support build approval based on repository settings [#1016](https://github.com/go-vela/server/commit/5fc317f98d5c56583cd19ecfd2e714747d5dd3cf) - thanks [@ecrupper](https://github.com/ecrupper)!
+- (types) feat(repo)!: change `allow_<event>` fields to an `AllowEvents` struct + DB use integer masking [#314](https://github.com/go-vela/types/commit/23c15858da2a56259b0cf6c2cf77aeb9e24d3f95) - thanks [@ecrupper](https://github.com/ecrupper)!
+
+#### Users:
+- `v0.23.x` introduces a new support build approval based on repository settings. Please take note of the [build approval information](/migrations/v0.23/build-approval.md).
+- `v0.23.x` introduces a new `deployments` table that will be created automatically by default. However, if you are running the `server` component with the `VELA_DATABASE_SKIP_CREATION` set to `true`, you will need to manually create this table.
```
Comment (ecrupper) [inline releases/v0.23.md:26]: This sentence should be in the `For Vela Administrators` section, but the bullet below should stay here in `Users`

File: releases/v0.23.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+- (server) feat(api/database)!: store deployment record in database for Vela-targeted deployments [#1031](https://github.com/go-vela/server/commit/a7cd07fe43cd7e5e0e79f7e1ddb6ca055d518d8a) - thanks [@claire1618](https://github.com/claire1618)!
+- (server) feat(repo)!: support allow_events struct [#1023](https://github.com/go-vela/server/commit/6fcde8652f63e317137a63cb4f01586bbbbc49c2) - thanks [@ecrupper](https://github.com/ecrupper)!
+- (server) feat(webhook)!: support build approval based on repository settings [#1016](https://github.com/go-vela/server/commit/5fc317f98d5c56583cd19ecfd2e714747d5dd3cf) - thanks [@ecrupper](https://github.com/ecrupper)!
+- (types) feat(repo)!: change `allow_<event>` fields to an `AllowEvents` struct + DB use integer masking [#314](https://github.com/go-vela/types/commit/23c15858da2a56259b0cf6c2cf77aeb9e24d3f95) - thanks [@ecrupper](https://github.com/ecrupper)!
+
+#### Users:
+- `v0.23.x` introduces a new support build approval based on repository settings. Please take note of the [build approval information](/migrations/v0.23/build-approval.md).
```
Comment (ecrupper) [inline releases/v0.23.md:25]:
```suggestion
- `v0.23.x` introduces build approval based on repository settings. Please take note of the [build approval information](/migrations/v0.23/build-approval.md).
```

**Review by ecrupper — CHANGES_REQUESTED** [review]

File: releases/v0.23.md (hunk 3 above)
Comment (timhuynh94) [reply releases/v0.23.md:26]: Done

File: releases/v0.23.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+- (server) feat(webhook)!: support build approval based on repository settings [#1016](https://github.com/go-vela/server/commit/5fc317f98d5c56583cd19ecfd2e714747d5dd3cf) - thanks [@ecrupper](https://github.com/ecrupper)!
+- (types) feat(repo)!: change `allow_<event>` fields to an `AllowEvents` struct + DB use integer masking [#314](https://github.com/go-vela/types/commit/23c15858da2a56259b0cf6c2cf77aeb9e24d3f95) - thanks [@ecrupper](https://github.com/ecrupper)!
+
+#### Admins:
+
+- `v0.23.x` introduces a new `deployments` table that will be created automatically by default. However, if you are running the `server` component with the `VELA_DATABASE_SKIP_CREATION` set to `true`, you will need to manually create this table.
+
+#### Users:
+- `v0.23.x` introduces build approval based on repository settings. Please take note of the [build approval information](/migrations/v0.23/build-approval.md).
+- Please take note of the [deployment recovery information](/migrations/v0.23/deploy-recovery.md).
```
Comment (wass3rw3rk) [inline releases/v0.23.md:30]:
```suggestion
- With the new deployment-related changes, please take note of the [deployment recovery information](/migrations/v0.23/deploy-recovery.md).
```

File: releases/v0.23.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+## For Vela Administrators
+
+- When upgrading from `v0.23`, please take note of the [migration information](/migrations/v0.23/README.md)
+
+## v0.23.0
+
+### 📣 Highlights
+
+
```
Comment (wass3rw3rk) [inline releases/v0.23.md:14]: we'll want to fill this in

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]

File: releases/v0.23.md (hunk 6 above)
Comment (timhuynh94) [reply releases/v0.23.md]: added!

File: releases/v0.23.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+### 📣 Highlights
+
+- Allowed Events
+- Deployment records
+- Improved Local Pipeline Execution Experience
+- Improved Pipeline Visualization
+- Support Build Approval
+- Bug Fixes and other Enhancements
+
```
Comment (wass3rw3rk) [inline releases/v0.23.md:20]:
```suggestion

- Allowed Events
- Native Deployment Records
- Introducing Build Approval
- Bug Fixes and other Enhancements

```

looks like some commits from previous release made it into the list below. let's rerun after final version is cut

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]

**Review by ecrupper — APPROVED** [review]

**Pull request merged**